### DISCUSSION TITLE
Scrum sprints--Stretch goals, feature enhancements

### DISCUSSION THREAD

**Discussion by jimguistical** [nss-evening-cohort-14/evening-client-side#133] [Discuss a Topic]
So we use the term "Stretch" for optional feature enhancements on our projects, and I'm curious what ya'll have found out in the real-world dev environment.  
For example, is there a general backlog for features and tickets that get incorporated as there is time?  
Or maybe any backlogs/stretch goals are resolved within a few sprints or are otherwise reworked into future a "feature enhancement."

Seems like the idea of stretch goals would be a difficult concept in real-time scenarios, especially if the client requests to make big changes to a product in future updates.

This article is what lead me to open this ticket for discussion:
https://www.scrum.org/forum/scrum-forum/5445/stretch-goals-good-or-bad

Thanks!

**Comment by drteresavasquez** [comment]
Semantics. Stretch in our usecase is a challenge for you to try after you've completed all the requirements of a project. 

We use backlog in the field but usually backlog items are things that have not yet been refined or fully fleshed out, but we know will be a feature of the application at some point. A flesh out would be both a prototype and a technical design which includes a spike (another new word) which is a discovery phase for what would be needed to do the work and is it feasible. 

Every ticket gets points, which basically is the time that it should take to complete the ticket. So when planning, the scrum master can identify how many average points can a team usually complete in a sprint. Some sprints are 1 week, others are 2-3 weeks depending on your company. 

As a junior dev when you finally get a ticket it will be refined and pointed for you, which also tells you the complexity of it. 

So, back to stretch. Stretch goals are things that you may be able to squeeze in so in a team environment we usually pull from the top of the refined tickets into the sprint and it becomes a sprint commitment when we do that, which essential is a stretch. It means that it was not planned for this sprint, but if we are able to pull it in and get it done, it was a stretch because it was not a part of the original commitment.

**Reply by jimguistical** [reply]
Thank you, Dr. T!  Agreed, seems like a lot of it is semantics depending on your team/company.  The idea of a "spike" is fascinating -- so you're pretty much trying to compartmentalize the risk-factor that comes with development a prototype feature?

**Reply by drteresavasquez** [reply]
That and trying to figure out how to implement the feature technically. So for a spike, you are taking time in a sprint that is time boxed. So a spike ticket may be pointed for 2 days of discovery and a tech analysis of how to implement a feature before actually starting on the dev ticket and refining it. 

Usually teams have a refinement meeting where a product manager and the devs point the tickets. The spike informs how many points the specific ticket will take to be completed and if it needs to be broken down into more tickets. 

So let's say you and I are on a team. Our average velocity (points we can complete) on a 2 week sprint is 15 points. We would not take on a tickets that when combined are more than 15 points. A spike can be a part of that sprint, which would mean one of us would be spending time doing technical discovery on a prospective feature so at the next refinement we can point the feature for the following sprint. You would not do both a spike and work on a feature at the same time. Does that make sense?

**Reply by drteresavasquez** [reply]
Also I LOVE this conversation!!! ❤️❤️

**Reply by jimguistical** [reply]
Me, too!  My agency is working with a 3rd-party Dev Team to build out a web-based grant management system, and a few jobs ago, I worked with a SalesForce consultant team who were helping our Non-profit partnership build out a SF Instance for case managers to share case notes, etc. across agencies... and I always wanted to know what the hell they were doing behind the scenes to organize all the feature updates we requested!  

It makes sense!  Where does the points average come from?  Feedback from "pointing" tickets for previous sprints, maybe?  And so the spike in our example... it's implied that we've decided along with Product Manager that this prospective feature will add value to our project.  Does it sometimes happen that a team decided thru a spike that it's not worth the time/$ to build out a prospective feature? Are things scrapped or "backlogged"?

Our Devs will be like -- are you sure you want to get rid of that feature totally for this new grant year?  We can delete it or we can just disable it and rework it later.  Sounds like they're squirreling away everything for easy access later.  (:

**Reply by drteresavasquez** [reply]
That number comes from past work the team has completed. The points depend on the org. I've worked with some that point 1-8 and some that do 1-5. Again depends on the org but typically a 1 is 1-3 hours, 2 is 3-6 hours, 3 is 1-3 days 5 is 3-5 days and 8 is 5-7 days and a 10 or above is more but also a sign that the ticket needs to be broken down into smaller tickets. All depends on the company.
